core/domain/activity_jobs_one_off.py

```diff
@@ -246,3 +246,230 @@ def reduce(key, values):
+    def map(snapshot_model):
+        """Implements the map function for this job."""
+        class_name = snapshot_model.__class__.__name__
+        if isinstance(
+                snapshot_model,
+                collection_models.CollectionRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_col_and_exp_user_ids(
+                snapshot_model)
+        elif isinstance(
+                snapshot_model,
+                exp_models.ExplorationRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_col_and_exp_user_ids(
+                snapshot_model)
+        elif isinstance(
+                snapshot_model,
+                topic_models.TopicRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_topic_user_ids(snapshot_model)
+        yield ('SUCCESS-%s' % class_name, snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, ids):
+        """Implements the reduce function for this job."""
+        yield (key, len(ids))
+
+
+class AuditSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that audits commit_cmds field of the snapshot metadata models. We log
+    the length of the commit_cmd, the possible 'cmd' values, and all the other
+    keys.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over three types of entity class.
+        super(AuditSnapshotMetadataModelsJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotMetadataModel,
+                exp_models.ExplorationRightsSnapshotMetadataModel,
+                topic_models.TopicRightsSnapshotMetadataModel]
+
+    @staticmethod
+    def map(snapshot_model):
+        """Implements the map function for this job."""
+        if isinstance(
+                snapshot_model,
+                collection_models.CollectionRightsSnapshotMetadataModel):
+            model_type_name = 'collection'
+        elif isinstance(
+                snapshot_model,
+                exp_models.ExplorationRightsSnapshotMetadataModel):
+            model_type_name = 'exploration'
+        elif isinstance(
+                snapshot_model,
+                topic_models.TopicRightsSnapshotMetadataModel):
+            model_type_name = 'topic'
+        yield (
+            '%s-length-%s' % (model_type_name, len(snapshot_model.commit_cmds)),
+            1
+        )
+        for commit_cmd in snapshot_model.commit_cmds:
+            for cmd_key, cmd_value in commit_cmd.items():
+                if cmd_key == 'cmd':
+                    yield ('%s-cmd-%s' % (model_type_name, cmd_value), 1)
+                else:
+                    yield ('%s-field-%s' % (model_type_name, cmd_key), 1)
```

Comment:
Perhaps worth including the cmd name here too, so that you know what fields correspond to what cmds?

Reply:
Done.